From ea8bcbf946a2dae6c390bc2b8d0566ba9207b9c6 Mon Sep 17 00:00:00 2001
From: Peter Griffin <peter.griffin@linaro.org>
Date: Thu, 23 Jan 2020 12:54:54 +0100
Subject: [PATCH 11/18] FindWestonClient.cmake: fix a few errors

weston-compositor isn't a .pc file it is called
weston.pc.

Also make adding the library conditional on having
found Weston in the system.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>
---
 cmake/modules/FindWestonClient.cmake | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/cmake/modules/FindWestonClient.cmake b/cmake/modules/FindWestonClient.cmake
index 22b68c3..f88b6f7 100644
--- a/cmake/modules/FindWestonClient.cmake
+++ b/cmake/modules/FindWestonClient.cmake
@@ -31,17 +31,15 @@
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 find_package(PkgConfig)
-pkg_check_modules(PC_WESTON weston-compositor)
+pkg_check_modules(PC_WESTON weston)
 
-find_library(WESTON_CLIENT_LIB NAMES weston-desktop-5
+find_library(WESTON_CLIENT_LIB NAMES weston-desktop-5 weston-desktop-6
         HINTS ${PC_WESTON_LIBDIR} ${PC_WESTON_LIBRARY_DIRS}
 )
 
 set (WESTON_CLIENT_LIBRARIES ${PC_WESTON_LIBRARIES})
 
-set (WESTON_CLIENT_FOUND TRUE)
-
-if(WESTON_CLIENT_FOUND AND NOT TARGET WestonClient::WestonClient)
+if(PC_WESTON_FOUND AND NOT TARGET WestonClient::WestonClient)
     set(WESTON_CLIENT_LIB_CLIENT_LINK_LIBRARIES "${WESTON_CLIENT_LIB}")
     add_library(WestonClient::WestonClient UNKNOWN IMPORTED)
     set_target_properties(WestonClient::WestonClient PROPERTIES
@@ -54,6 +52,6 @@ if(WESTON_CLIENT_FOUND AND NOT TARGET WestonClient::WestonClient)
 endif()
 
 include(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(PC_WESTON DEFAULT_MSG WESTON_CLIENT_FOUND)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(PC_WESTON DEFAULT_MSG PC_WESTON_FOUND)
 
 mark_as_advanced(WESTON_CLIENT_LIBRARIES)
-- 
2.17.1

